### PR TITLE
WIP: Add a "materialize" command

### PR DESCRIPTION
**Pull request by dimaryaz**
Takes either a package name or a node.

TODO: Is this the API we want? Alternative: a `_materialize()` method on a node.

### CODE REVIEW

File: compiler/quilt/tools/command.py [hunk 1]
```diff
@@ -940,6 +940,53 @@ def install(package, hash=None, version=None, tag=None, force=False, meta_only=F
...
+        node = package
+    else:
+        raise ValueError("Expected a Node or a package name, but got %r" % package)
+
+    store = PackageStore()
+
+    hashes = set()
+
+    stack = [node]
+    while stack:
```
Comment (kevinemoore) [inline compiler/quilt/tools/command.py]: This is reimplementing tree traversal (we now have a helper, right?) and is duplicating the fragment download logic from install. Maybe it'll look cleaner after merging your data_transfer changes?

Comment (dimaryaz) [reply compiler/quilt/tools/command.py]: The helper is for core nodes (which we don't have until we "build" the package... yeah, I know, unnecessarily confusing.)

As for duplicating: where do you see that? My data_transfer changes _are_ merged.

**Comment by kevinemoore** [comment]
Both install and materialize start with a list of hashes, then strip away the ones already installed. Install gets that list via the database, but that's basically a shortcut around downloading and parsing the package manifest. 

It seems strange to me now that install automatically upgrades the package version. So if a user does:
quilt install foo/bar/subpath  (waits while a new version is pushed, then)
quilt install foo/bar
They'll get prompted to either overwrite (and upgrade) or cancel all together. Materialize seems like a more logical behavior there.

**Comment by dimaryaz** [comment]
Ok sure. I'll try that.

**Pull request closed without merging**